src/service/video-manager-impl.js

```diff
@@ -217,21 +224,56 @@ class VideoEntry {
       // Only muted videos are allowed to autoplay
       this.video.mute();
 
-      // If autoplay video has controls, hide them and only show them on
-      // user interaction.
-      if (this.video.element.hasAttribute(VideoAttributes.CONTROLS)) {
-        this.video.hideControls();
-
-        listenOnce(this.video.element, VideoEvents.USER_TAP, () => {
-          this.userInteracted_ = true;
-          this.video.showControls();
-          this.video.unmute();
-        });
+      if (this.isInteractive_) {
+        this.autoplayInteractiveVideoBuilt_();
       }
     });
   }
 
   /**
+   * Called by autoplayVideoBuilt_ when an interactive autoplay video is built.
+   * It handles hiding controls, installing autoplay animation and handling
+   * user interaction by unmuting and showing controls.
+   * @private
+   */
+  autoplayInteractiveVideoBuilt_() {
+    const onInteraction = () => {
+      this.userInteracted_ = true;
+      this.video.showControls();
+      this.video.unmute();
+      unlistenInteraction();
```

Comment:
Danger! Is is not valid to reference a const (or let) variable before it is declared! You could rewrite this into a FunctionDeclaration (instead of an ArrowFunctionExpression) and place it at the end of the method (function hoisting will take care of the rest):

```js
function autoplayInteractiveVideoBuilt_() {
  const toggleAnimation...
  const unlistenInteraction = listen(this.video.element, VideoEvents.USER_TAP, onInteraction.bind(this));
  const unlistenPause...
  const unlistenPlay...

  function onInteraction() {
    // I can be used anywhere in autoplayInteractiveVideoBuilt_
  }
}